kart/merge_util.py

```diff
@@ -819,3 +830,105 @@ def merge_status_to_text(jdict, fresh):
+        do our best to write it to the working copy anyway, so that the user can see those parts that merged cleanly,
+        and those conflicts that we can write to the WC, and so that they can use the WC as a starting point for
+        specifying resolves.
+        """
+        # First pass - forcibly resolve conflicts in the merge-index, write to a tree, then write that to the WC:
+        tree_id = merge_index.write_resolved_tree(self.repo, self.conflict_resolver)
+        self.repo.working_copy.reset(self.repo[tree_id], quiet=True)
+        # Second pass - where possible, handle conflicts that can be written in a more complicated way without resolving them:
+        self.write_conflicts_to_working_copy(merge_index)
+
+    def conflict_resolver(self, conflict):
+        """
+        First pass - we forcibly but somewhat arbitrarily resolve any outstanding conflicts in the merge-index,
+        write the result to tree, and use that tree to update the working copy.
+        """
+        rich_conflict = RichConflict(conflict, self.merge_context)
+        # For feature and meta conflicts we can really only fit one possibility in the working copy.
+        # We go with the "ours" versions, until the user specifies something else.
+        dataset_part = rich_conflict.decoded_path[1]
+        if not rich_conflict.has_multiple_paths and dataset_part in ("meta", "feature"):
+            ours = rich_conflict.versions.ours
+            if ours:
+                yield ours.entry
+        # Tile conflicts are effectively resolved as "deleted" here since we don't yield any resolve for them.
+        # They are handled below in "write_conflicts_to_working_copy"
+
+    def write_conflicts_to_working_copy(self, merge_index):
+        """
+        Second pass - those conflicts that can be written to the working copy in a more complicated way (ie
+        without resolving them one way or another) are handled here.
+        """
+        # So far we only write conflicts to the workdir - writing other types conflicts to the working copy
+        # (to the extent possible) is handled in the first pass by the conflict resolver.
+        if not self.repo.working_copy.workdir:
+            return
+
+        for conflict in merge_index.unresolved_conflicts.values():
+            rich_conflict = RichConflict(conflict, self.merge_context)
+            dataset_path, dataset_part, item_path = rich_conflict.decoded_path
+
+            if not rich_conflict.has_multiple_paths and dataset_part == "tile":
+                # Tile conflict found - write all versions we can find to the workdir, with different names.
+                tilename = item_path
+                from kart.lfs_util import (
+                    pointer_file_bytes_to_dict,
+                    get_local_path_from_lfs_hash,
+                )
+                from kart.point_cloud.tilename_util import set_tile_extension
+
+                self._ensure_lfs_tiles_fetched()
```

Comment:
Oooh good catch - I thought I wanted to only do this in case of encountering a tile conflict, in which case, I need to fetch all possibilities so I can maybe put them in the conflict.
But, I think I need do this even before the first pass, so that if a tile cleanly merges to one from a different commit, that will be ready to write to the WC too...